### PR TITLE
 Move typing animation to above the input on mobile.

### PR DESCRIPTION
**Pull request by adamjspooner**
@keybase/react-hackers, this is ready for review.

### CODE REVIEW

**Comment by mmaxim** [comment]
Is it still just dots, or can you see the name?

**Comment by adamjspooner** [comment]
You can see the names on mobile now.

**Comment by adamjspooner** [comment]
@keybase/react-hackers, this is ready for another look.

File: shared/chat/conversation/input-area/normal/typing/container.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+import * as Constants from '../../../../../constants/chat2'
+import * as Types from '../../../../../constants/types/chat2'
+import {connect} from '../../../../../util/container'
+import {Typing} from '.'
+
+type OwnProps = {|
+  conversationIDKey: Types.ConversationIDKey,
+|}
+
+const mapStateToProps = (state, {conversationIDKey}: OwnProps) => {
```
Comment (chrisnojima) [inline shared/chat/conversation/input-area/normal/typing/container.js]: can be
```
=> ({names: Constants.getTyping(state, conversationIDKey)})
```

🐑

File: shared/chat/conversation/input-area/normal/typing/index.js [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+        </>
+      )
+      break
+    default:
+      ret = 'multiple people are typing'
+  }
+  return ret
+}
+
+export type Props = {|
```
Comment (chrisnojima) [inline shared/chat/conversation/input-area/normal/typing/index.js]: don't need to export this

File: shared/chat/conversation/input-area/normal/typing/index.js [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+    style={Styles.collapseStyles([
+      styles.isTypingContainer,
+      props.names && props.names.size > 0 && styles.isTypingContainerVisible,
+    ])}
+  >
+    <Kb.Box style={styles.typingIconContainer}>
+      <Kb.Animation animationType="typing" containerStyle={styles.isTypingAnimation} />
+    </Kb.Box>
+    <Kb.Text type={Styles.isMobile ? 'BodyTiny' : 'BodySmall'} style={styles.isTypingText}>
+      <Names names={props.names ?? I.Set()} />
```
Comment (chrisnojima) [inline shared/chat/conversation/input-area/normal/typing/index.js]: why do this, just have the container do it

File: shared/chat/conversation/input-area/normal/typing/index.js [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+      break
+    default:
+      ret = 'multiple people are typing'
+  }
+  return ret
+}
+
+export type Props = {|
+  conversationIDKey: Types.ConversationIDKey,
+  names?: I.Set<string>,
```
Comment (chrisnojima) [inline shared/chat/conversation/input-area/normal/typing/index.js]: this isn't optional, it always comes

File: shared/chat/conversation/input-area/normal/typing/index.js [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+      )
+      break
+    default:
+      ret = 'multiple people are typing'
+  }
+  return ret
+}
+
+export type Props = {|
+  conversationIDKey: Types.ConversationIDKey,
```
Comment (chrisnojima) [inline shared/chat/conversation/input-area/normal/typing/index.js]: this isn't a prop it needs

File: shared/chat/conversation/input-area/normal/typing/index.js [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+        </>
+      )
+      break
+    default:
+      ret = 'multiple people are typing'
+  }
+  return ret
+}
+
+type Props = {
```
Comment (chrisnojima) [inline shared/chat/conversation/input-area/normal/typing/index.js]: should be exact

Comment (adamjspooner) [reply shared/chat/conversation/input-area/normal/typing/index.js]: I'm not sure how to do that and make the stories happy. The component has exactly `names` and the connected component has exactly `conversationIDKey`. Do you have any suggestions?

Comment (mmaxim) [reply shared/chat/conversation/input-area/normal/typing/index.js]: I put in a commit to fix it.

**Comment by adamjspooner** [comment]
Thanks, @mmaxim!

**Pull request merged**